pkg/operators/kubenameresolver/kubenameresolver.go

```diff
@@ -134,6 +138,150 @@ func (m *KubeNameResolverInstance) EnrichEvent(ev any) error {
 	return nil
 }
 
+func (k *KubeNameResolver) GlobalParams() api.Params {
+	return nil
+}
+
+func (k *KubeNameResolver) InstanceParams() api.Params {
+	return nil
+}
+
+func (k *KubeNameResolver) Priority() int {
+	return priority
+}
+
+type k8sAccesors struct {
+	PodName   datasource.FieldAccessor
+	Namespace datasource.FieldAccessor
+	PodIP     datasource.FieldAccessor
+	HostIP    datasource.FieldAccessor
+	PodLabels datasource.FieldAccessor
+}
+
+func (k *KubeNameResolver) InstantiateDataOperator(gadgetCtx operators.GadgetContext, instanceParamValues api.ParamValues) (operators.DataOperatorInstance, error) {
+	logger := gadgetCtx.Logger()
+	accessors := make(map[datasource.DataSource]k8sAccesors)
+	for _, ds := range gadgetCtx.GetDataSources() {
+		logger.Debugf("KubeNameResolverOperator inspecting datasource %q", ds.Name())
+
+		if enable, ok := ds.Annotations()["kubenameresolver.enable"]; !ok || enable != "true" {
+			logger.Debugf("KubeNameResolverOperator not enabled by annotation")
+			continue
+		}
+
+		k8sField := ds.GetField("k8s")
+		if k8sField == nil {
+			logger.Debugf("> no k8s fields found")
+			continue
+		}
+
+		k8sAccesors := k8sAccesors{}
+		var err error
+
+		k8sAccesors.PodName = ds.GetField("k8s.podName")
+		if k8sAccesors.PodName == nil {
+			logger.Warnf("No podName field found in datasource %q", ds.Name())
+			continue
+		}
+
+		k8sAccesors.Namespace = ds.GetField("k8s.namespace")
+		if k8sAccesors.Namespace == nil {
+			logger.Warnf("No namespace field found in datasource %q", ds.Name())
+			continue
+		}
+
+		k8sAccesors.PodLabels = ds.GetField("k8s.podLabels")
+		if k8sAccesors.PodLabels == nil {
+			logger.Warnf("No podLabels field found in datasource %q", ds.Name())
+			continue
+		}
```

Comment:
I can't understand how PodLabels is used. I know PodName and Namespace are used to lookup the pod, but this field is not needed there. So what's the relationship of this operator and this field? I mean,  it's created and set on pkg/datasource/compat/wrapper.go, so why is this operator touching it?

Reply:
I guess this operator preceeded the changes that led to `PodLabels` be enriched by default in `wrapper.go`.

You are right, this is not needed anymore. Removing it